File: Data Structures & Algorithms/eating-bananas/submission-2.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        L, R = 1, max(piles)
        
        while L <= R:
            k = (R + L) // 2
            total = 0

            for p in piles:
                total += math.ceil(p / k)

            if total <= h:
                R = k - 1
                res = k
            else:
                L = k + 1

        return res
            

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int r = prices.size()-1;
        int res = 0;

        for(int l = 0; l < prices.size(); ++l){
            r = prices.size()-1;
            while(l < r){
                if(prices[l] <= prices[r]){
                    res = max(res, prices[r] - prices[l]);
                }

                --r;
            }
        }

        return res;
    }
};
